homeassistant/components/workday/binary_sensor.py

```diff
@@ -76,21 +81,47 @@ def valid_country(value: Any) -> str:
 )
 
 
-def setup_platform(
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
-    add_entities: AddEntitiesCallback,
+    async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Workday sensor."""
-    add_holidays: list[DateLike] = config[CONF_ADD_HOLIDAYS]
-    remove_holidays: list[str] = config[CONF_REMOVE_HOLIDAYS]
-    country: str = config[CONF_COUNTRY]
-    days_offset: int = config[CONF_OFFSET]
-    excludes: list[str] = config[CONF_EXCLUDES]
-    province: str | None = config.get(CONF_PROVINCE)
-    sensor_name: str = config[CONF_NAME]
-    workdays: list[str] = config[CONF_WORKDAYS]
+    async_create_issue(
+        hass,
+        DOMAIN,
+        "deprecated_yaml",
+        breaks_in_ha_version="2023.7.0",
+        is_fixable=False,
+        severity=IssueSeverity.WARNING,
+        translation_key="deprecated_yaml",
+    )
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={
+                **config,
+                CONF_PROVINCE: config.get(CONF_PROVINCE),
+            },
```

Comment:
I think it's better to make the adjustments inside the import step.
```suggestion
            data=config,
```